client/components/data/document/index.jsx

```diff
@@ -0,0 +1,100 @@
+/**
+ * External dependencies.
+ */
+import { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import each from 'lodash/each';
+
+/**
+ * Internal dependencies.
+ */
+import {
+	setTitle,
+	setDescription,
+	addLink,
+	addMeta,
+	setUnreadCount
+} from 'state/document/actions';
+
+class Document extends Component {
+	componentWillMount() {
+		this.props.setTitle( this.props.title );
+		this.props.setDescription( this.props.description );
+		this.props.setUnreadCount( this.props.unreadCount );
+
+		if ( this.props.link ) {
+			each( this.props.link, ( link ) => {
+				this.props.addLink( link );
```

Comment:
We could overload `addLink` or implement a separate `addLinks` action creator for convenience, since I imagine this will be a common pattern.